tests/foreman/cli/test_reporttemplates.py

```diff
@@ -857,3 +857,41 @@ def test_positive_generate_entitlements_report_multiple_formats(self):
             })
             assert vm.hostname in result_csv[1]
             assert self.setup_subs_id[0]['name'] in result_csv[1]
+
+    @tier3
+    def test_positive_schedule_Entitlements_report(self):
+        """Schedule an report using the Entitlements template in csv format.
+
+        :id: 572fb387-86e0-40e2-b2df-e8ec26433610
+
+
+        :setup: Installed Satellite with Organization, Activation key,
+                Content View, Content Host, and Subscriptions.
+
+        :steps:
+            1. hammer report-template schedule --organization '' --id '' --report-format ''
+
+        :expectedresults: report is scheduled and generated containing all the expected information
+                          regarding Entitlements.
+
+        :CaseImportance: High
+        """
+        with VirtualMachine(distro=DISTRO_RHEL7) as vm:
+            vm.install_katello_ca()
+            vm.register_contenthost(self.setup_org['label'], self.setup_new_ak['name'])
+            assert vm.subscribed
+            scheduled_csv = ReportTemplate.schedule({
+                'id': 115,
+                'organization': self.setup_org['name'],
+                'report-format': 'csv'
+            })
+            data_csv = ReportTemplate.report_data({
+                'id': 115,
+                'job-id': scheduled_csv[0].split("Job ID: ", 1)[1]
+            })
+            for item in data_csv:
+                if vm.hostname == item:
+                    assert vm.hostname in data_csv
```

Comment:
I have a query here if the` if vm.hostname == item` False over the loop we don't have the assertion in place ?
why not directly using `assert vm.hostname in data_csv ` maybe I'm missing something ?

Reply:
report_data has a unique return value. It is a list with multiple strings, each containing multiple values. If I try to `assert vm.hostname in data_cav` it raises an assertion error. Hence me looping through instead.

Reply:
1. If `assert vm.hostname in data_csv` failed outside of loop, there is no reason to believe it will succeed inside loop.
2. That assert will never be executed, because `item` never contains only `vm.hostname` - there are multiple other things in line.

What you might have meant (untested):
```python
for item in data_csv:
    if vm.hostname in item:
        assert vm.hostname in item
        break
else:  # this will be executed only if loop body did not encounter `break`
    # we need, because if vm.hostname is not in file anywhere, we don't run any assertion and we never fail
    pytest.fail("VM hostname {} not found in CSV".format(vm.hostname)
```

But it's much better written this way:
```python
assert any(vm.hostname in line for line in data_csv)
```

Alternatively, you can call `csv.reader()` on `data_csv` (`reader` supports list of strings as well as any file-like object) and check if `vm.hostname` is equal to specific field in any line.